File: src/Array/T0475_Heaters.java
package Array;

import java.util.Arrays;

public class T0475_Heaters {
    public int findRadius(int[] houses, int[] heaters) {
        int max = 0,len = 0;
        int i,j,site;
        Arrays.sort(houses);
        Arrays.sort(heaters);
        for(i = 0 ;i < houses.length;i++){
            site = houses[i];
            len = Math.abs(heaters[0] - site);
            for(j = 0 ;j < heaters.length;j++){
                if(len > Math.abs(heaters[j]-site))
                    len = Math.abs(heaters[j]-site);
            }
            max = Math.max(len,max);
        }
        return max;
    }
}

File: src_swift/T0475_Heaters.swift
class T0475_Heaters{
    func findRadius(_ houses: [Int], _ heaters: [Int]) -> Int {
        let houses = houses.sorted()
        let heaters = heaters.sorted()
        var ans = 0,len = 0
        for i in 0..<houses.count{
            var site = houses[i]
            len = abs(heaters[0] - site)
            for j in 0..<heaters.count{
                if len > abs(heaters[j]-site){
                    len = abs(heaters[j]-site)
                }
            }
            ans = max(len,ans)
        }
        return ans
    }
}